src/Corax/IndexSearcher/IndexSearcher.Search.cs

```diff
@@ -34,7 +34,7 @@ public IQueryMatch SearchQuery(string field, string searchTerm, Constants.Search
             };
         }
 
-        if (_fieldMapping.TryGetByFieldId(analyzerId, out var indexFieldBinding) == false && indexFieldBinding.Analyzer is null && isDynamic == false)
+        if ((_fieldMapping.TryGetByFieldId(analyzerId, out var indexFieldBinding) == false || indexFieldBinding.Analyzer is null) && isDynamic == false)
         {
             throw new InvalidOperationException($"{nameof(SearchQuery)} requires analyzer.");
```

Comment:
Shouldn't we add the field name here?
And don't we have a default analyzer?

Reply:
We can, but if a field is not mapped or it is not dynamic (in such case fieldId is `-2`) there's something wrong. We have a default analyzer, but it is mentioned only for dynamic fields. There are tests that you can change the analyzer during `where search()` for known fields.